api/v1alpha3/cluster_types.go

```diff
@@ -18,14 +18,16 @@ package v1alpha3
 
 import (
 	"fmt"
-
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	capierrors "sigs.k8s.io/cluster-api/errors"
 )
 
 const (
 	ClusterFinalizer = "cluster.cluster.x-k8s.io"
+
+	// PostApplyAnnotationPrefix is the prefix of the annotations that is used to determine if an addon is applied or not.
+	PostApplyAnnotationPrefix = "cluster.x-k8s.io/postapply-secret"
```

Comment:
I'd rather not overload the Spec here if we don't need to, unless we expect users to also set the field. Otherwise, I think we can track it in Status with a separate struct.

Reply:
I thought of that, but status will be lost in case we pivot/move or backup/restore. I definitely prefer another field than having a bunch of dynamically generated annotations to be set on the Cluster